pkgs/by-name/pa/parallel-disk-usage/package.nix

```diff
@@ -0,0 +1,28 @@
+{
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "parallel-disk-usage";
+  version = "0.9.0";
+
+  src = fetchFromGitHub {
+    owner = "KSXGitHub";
+    repo = pname;
+    rev = version;
+    hash = "sha256-kOMbVKwnGh47zZkWAWkctfTIE5F8oeSnAgJEU/OdsQc=";
+  };
+
+  cargoHash = "sha256-Jk9sNvApq4t/FoEzfjlDT2Td5sr38Jbdo6RoaOVQJK8=";
+
+  meta = with lib; {
+    description = "Highly parallelized, blazing fast directory tree analyzer";
+    homepage = "https://github.com/KSXGitHub/parallel-disk-usage";
+    license = licenses.asl20;
+    maintainers = [maintainers.peret];
+    mainProgram = "pdu";
+    platforms = platforms.all;
+    badPlatforms = ["aarch64-linux"];
```

Comment:
the unit test failure looks like an issue with the test and not the platform and seems to be due to an ordering issue of the nested and flat nodes. an issue upstream needs to be filed.
```
---- multiple_names stdout ----
ACTUAL:
 6          ┌──1 │███████████████████▒▒▒▒▒▒▒░░░░░░                                             │ 25%
 8        ┌─┴0   │██████████████████████████░░░░░░                                             │ 33%
10      ┌─┴nested│████████████████████████████████                                             │ 42%
 1      │ ┌──1   │███░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░                                │  4%
 2      │ ├──2   │██████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░                                │  8%
 3      │ ├──3   │██████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░                                │ 13%
14      ├─┴flat  │█████████████████████████████████████████████                                │ 58%
24    ┌─┴(total) │█████████████████████████████████████████████████████████████████████████████│100%
EXPECTED:
 1        ┌──1   │███░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░                                │  4%
 2        ├──2   │██████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░                                │  8%
 3        ├──3   │██████████░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░                                │ 13%
14      ┌─┴flat  │█████████████████████████████████████████████                                │ 58%
 6      │   ┌──1 │███████████████████▒▒▒▒▒▒▒░░░░░░                                             │ 25%
 8      │ ┌─┴0   │██████████████████████████░░░░░░                                             │ 33%
10      ├─┴nested│████████████████████████████████                                             │ 42%
24    ┌─┴(total) │█████████████████████████████████████████████████████████████████████████████│100%
```

Reply:
since this is a failing unit test and aarch64-linux should work use:
```nix
broken = (stdenv.isLinux && stdenv.isAarch64);
```
rather than `badPlatforms`.

Reply:
and remove `platforms = platforms.all;` as that is not correct. leaving it blank will provide the default set of rust platforms, which is a subset of `platforms.all`.

Reply:
Yes, my assumption was that the test case might fail on aarch64-linux only, so this was my attempt to work around that.

Thank you for the review! I will make the suggested changes and raise an issue in the original repo.